homeassistant/components/ezviz/alarm_control_panel.py

```diff
@@ -0,0 +1,165 @@
+    """Representation of an Ezviz alarm control panel."""
+
+    entity_description: EzvizAlarmControlPanelEntityDescription
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_supported_features = (
+        AlarmControlPanelEntityFeature.ARM_AWAY
+        | AlarmControlPanelEntityFeature.ARM_HOME
+    )
+    _attr_code_arm_required = False
+
+    def __init__(
+        self,
+        coordinator: EzvizDataUpdateCoordinator,
+        entry_id: str,
+        device_info: DeviceInfo,
+        entity_description: EzvizAlarmControlPanelEntityDescription,
+    ) -> None:
+        """Initialize alarm control panel entity."""
+        self._attr_unique_id = f"{entry_id}_{entity_description.key}"
+        self._attr_device_info = device_info
+        self.entity_description = entity_description
+        self.coordinator = coordinator
+        self._attr_state = None
+
+    async def async_added_to_hass(self) -> None:
+        """Entity added to hass."""
+        self.schedule_update_ha_state(True)
+
+    def alarm_disarm(self, code: str | None = None) -> None:
+        """Send disarm command."""
+        try:
+            if self.coordinator.ezviz_client.api_set_defence_mode(
+                DefenseModeType.HOME_MODE.value
+            ):
+                self._attr_state = STATE_ALARM_DISARMED
+
+        except PyEzvizError as err:
+            raise HomeAssistantError("Cannot disarm EZVIZ alarm") from err
+
+    def alarm_arm_away(self, code: str | None = None) -> None:
+        """Send arm away command."""
+        try:
+            if self.coordinator.ezviz_client.api_set_defence_mode(
+                DefenseModeType.AWAY_MODE.value
+            ):
+                self._attr_state = STATE_ALARM_ARMED_AWAY
+
+        except PyEzvizError as err:
+            raise HomeAssistantError("Cannot arm EZVIZ alarm") from err
+
+    def alarm_arm_home(self, code: str | None = None) -> None:
+        """Send arm home command."""
+        try:
+            if self.coordinator.ezviz_client.api_set_defence_mode(
+                DefenseModeType.SLEEP_MODE.value
+            ):
+                self._attr_state = STATE_ALARM_ARMED_HOME
+
+        except PyEzvizError as err:
+            raise HomeAssistantError("Cannot arm EZVIZ alarm") from err
+
+    def update(self) -> None:
+        """Fetch data from EZVIZ."""
+        ezviz_alarm_state_number = "0"
+        try:
+            ezviz_alarm_state_number = (
+                self.coordinator.ezviz_client.get_group_defence_mode()
+            )
+            _LOGGER.debug(
+                "Updating EZVIZ alarm with response %s", ezviz_alarm_state_number
+            )
+            self._attr_state = self.entity_description.ezviz_alarm_states[
+                int(ezviz_alarm_state_number)
+            ]
+
+        except PyEzvizError as error:
+            raise HomeAssistantError(
+                f"Could not fetch EZVIZ alarm status: {error}"
+            ) from error
```

Comment:
Why is this not making use of the coordinator? Seems it would make more sense to let the coordinator handle updates for this entity type too?

Reply:
Hi,

Because it's a single entity with no reason to share data (only one per ezviz account) I thought it better to not use the coordinator. Also allows the user to disable this entity if they so choose (if disabled won't poll this endpoint)

Guess from an efficiency standpoint I though it better to do it this way. Have no problem implementing it in the coordinator if you think it's better?

Reply:
Its the library safe to use from multiple threads? Does the server handle multiple parallel connections? I'm asking since different platforms will update in parallel.

I think using the coordinator would be better. You can use coordinator contexts to limit what the coordinator fetch depending on which entities are active.

Gardena bluetooth have one example. But you can do it simpler. Just set self.coordinator_context to "alarm" in the alarm entitites, then check for "alarm" in self.async_contexts() in the coordinator update method.

Reply:
Im okey, with that being a separate refactor at some stage. So you don't need to do it now.

Reply:
The API can handle multiple API calls simultaneously so that won't be a problem.

I like the consistency of having everything in the coordinator so will definitely look into refactoring this.